### PR TITLE
Adding VirtualCluster related cmdlets and corresponding tests...

### PR DESCRIPTION
**Pull request by milostod**
...and help files.

## Description

This change adds cmdlets related to VirtualCluster resource. The intention is to enable customers to get information about virtual cluster(s) under their subscription and resource group, as well as to remove specified virtual cluster, which was a major ask from our customers. Explicit creation of new virtual clusters is not supported, as virtual cluster resource is only implicitly created when customer creates first managed instance in the specified subnet. Similarly, updates on virtual clusters are not currently supported.
Test coverage and recordings has been added for all chenges, as well as markdown help files.
Cmdlet design review is completed here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/271

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql.Test/Sql.Test.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.28.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.29.0-preview" />
```
Comment (milostod) [inline src/Sql/Sql.Test/Sql.Test.csproj]: Note: New NuGet package has been published from the Azure SDK for .NET repository

**Comment by milostod** [comment]
Note: there are some auto-generated changes in Az.Sql.md related to sensitivity classification/recommendation, as well as in Set-AzSqlDatabaseThreatDetectionPolicy.md and Set-AzSqlServerThreatDetectionPolicy.md. My guess is that original changes in these areas were not accompanied with md update through platyPS, so I included those in my change.

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/Sql/Sql/VirtualCluster/Cmdlet/GetAzureSqlVirtualCluster.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        /// <summary>
+        /// Gets or sets the name of the resource group.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [Parameter(ParameterSetName = GetByResourceGroupParameterSet,
+            Mandatory = false,
+            Position = 1,
```
Comment (cormacpayne) [inline src/Sql/Sql/VirtualCluster/Cmdlet/GetAzureSqlVirtualCluster.cs]: @milostod please remove the `Position` property from optional parameters

File: src/Sql/Sql/VirtualCluster/Cmdlet/RemoveAzureSqlVirtualCluster.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        [ValidateNotNullOrEmpty]
+        [Alias("VirtualCluster")]
+        public AzureSqlVirtualClusterModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource id of the virtual cluster
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveByResourceIdParameterSet,
+            Mandatory = true,
+            Position = 0,
```
Comment (cormacpayne) [inline src/Sql/Sql/VirtualCluster/Cmdlet/RemoveAzureSqlVirtualCluster.cs]: @milostod minor: you can remove this positional parameter since users will never be able to use it (if they provide a single string, PowerShell won't be able to differentiate between the Name/ResourceGroupName and ResourceId parameter sets, and since the Name/ResourceGroupName set is the default, it will choose that one and prompt the user to provide a value for `-ResourceGroupName`)

File: src/Sql/Sql/VirtualCluster/Services/AzureSqlVirtualClusterAdapter.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        /// <param name="resp">The management client Virtual Cluster response to convert</param>
+        /// <returns>The converted Virtual Cluster model</returns>
+        private static AzureSqlVirtualClusterModel CreateVirtualClusterModelFromResponse(Management.Sql.Models.VirtualCluster resp)
+        {
+            AzureSqlVirtualClusterModel virtualCluster = new AzureSqlVirtualClusterModel();
+
+            // Extract the resource group name from the ID.
+            // ID is in the form:
+            // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.Sql/virtualClusters/virtualClusterName
+            string[] segments = resp.Id.Split('/');
```
Comment (cormacpayne) [inline src/Sql/Sql/VirtualCluster/Services/AzureSqlVirtualClusterAdapter.cs]: @milostod you can use the `ResourceIdentifier` class that's in our common ARM code to safely parse a provided resource id string

File: src/Sql/Sql/VirtualCluster/Cmdlet/GetAzureSqlVirtualCluster.cs (hunk 2 above)
Comment (milostod) [reply src/Sql/Sql/VirtualCluster/Cmdlet/GetAzureSqlVirtualCluster.cs]: Done.

File: src/Sql/Sql/VirtualCluster/Cmdlet/RemoveAzureSqlVirtualCluster.cs (hunk 3 above)
Comment (milostod) [reply src/Sql/Sql/VirtualCluster/Cmdlet/RemoveAzureSqlVirtualCluster.cs]: I see, thanks. I've made the change.

File: src/Sql/Sql/VirtualCluster/Services/AzureSqlVirtualClusterAdapter.cs (hunk 4 above)
Comment (milostod) [reply src/Sql/Sql/VirtualCluster/Services/AzureSqlVirtualClusterAdapter.cs]: Nice, thanks for the comment.

**Comment by milostod** [comment]
@cormacpayne I've made changes we discussed. Windows and Linux builds failed in automated checks, although build works fine on my local machine. It seems I cannot request rebuild.

**Comment by milostod** [comment]
@cormacpayne @maddieclayton, I've made changes and fixed merge conflicts, please review.

File: src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1 [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+<#
+	.SYNOPSIS
+	Tests Getting a VirtualCluster
+	.DESCRIPTION
+	SmokeTest
+#>
+function Test-GetVirtualCluster
+{
+	# Setup
+	$location = "eastus"
```
Comment (maddieclayton) [inline src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: Please get location using Get-Location to ensure this will work on all clouds: https://github.com/Azure/azure-powershell/blob/master/tools/ScenarioTest.ResourceManager/Common.ps1#L583

File: src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1 [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	$rg = Create-ResourceGroupForTest $location
+
+	$rgName = "RG_MIPlayground"
```
Comment (maddieclayton) [inline src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: Please have all resource name be randomly generated using getAssetName: https://github.com/Azure/azure-powershell/blob/master/tools/ScenarioTest.ResourceManager/Common.ps1#L371

File: src/Sql/Sql/Az.Sql.psd1 [hunk 7]
```diff
@@ -271,7 +272,7 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Support Database Data Classification.'
+        ReleaseNotes = '* Add Virtual Cluster cmdlets'
```
Comment (maddieclayton) [inline src/Sql/Sql/Az.Sql.psd1]: Please revert this change, we will update during versioning.

File: src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1 (hunk 5 above)
Comment (milostod) [reply src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: Hi @maddieclayton , I used some pre-made resources to speed up the test because provisioning of a new virtual cluster takes several hours. I've made the changes to create everything in the test, however, each test takes almost 4 hours to execute. If this is not a problem, I'll commit the change. Please let me know.

Comment (maddieclayton) [reply src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: @milostod It is fine to take 4 hours to record this test - because the SDK calls will be instantaneous in playback mode, it will not affect our CI, and having a test take 4 hours to record is much better than having to do manual set up that takes the same amount of time.

Comment (milostod) [reply src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: @maddieclayton I agree. I've made the changes, tests are now creating everything from scratch, including ResourceGroup, VNet and Subnet. Total running time was 7hrs:30min for two tests. I used Get-ProviderLocation since it uses Get-AzResourceProvider instead of Get-AzureRmResourceProvider command for which I had some trouble running locally.

File: src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1 (hunk 6 above)
Comment (milostod) [reply src/Sql/Sql.Test/ScenarioTests/VirtualClusterCrudScenarioTests.ps1]: Done. rgName is created in Create-ResourceGroupForTest using getAssetName. I kept hardcoded names for vnet and subnet as it seems that calling getAssetName multiple times in short period produces "queue is empty" error. I see multiple tests are keeping hardcoded values for vnet and subnet, probably for the same reason.

File: src/Sql/Sql/Az.Sql.psd1 (hunk 7 above)
Comment (milostod) [reply src/Sql/Sql/Az.Sql.psd1]: Done, thanks.
